fvm/src/executor/default.rs

```diff
@@ -123,17 +124,52 @@ where
             });
 
             let result = cm.with_transaction(false, |cm| {
+                let pl = cm.context().price_list;
+
+                if let Some(sender_state) = sender_st {
+                    if cm
+                        .machine()
+                        .builtin_actors()
+                        .is_eoa_actor(&sender_state.code)
+                    {
+                        // We need to deploy the Ethereum EOA actor
+                        cm.charge_gas(pl.on_create_actor(false))?;
+
+                        // Transform the actor code CID into the Ethereum EOA code CID.
+                        let eoa_code_cid = *cm.machine().builtin_actors().get_eoa_code();
+                        cm.state_tree_mut().set_actor(
+                            sender_id,
+                            ActorState {
+                                code: eoa_code_cid,
+                                ..sender_state
+                            },
+                        )?;
+
+                        // Now actually invoke the constructor
+                        let ret = cm.send::<K>(
```

Comment:
Hm. Now I see why you wanted this inside the call manager.

For now, I'd consider just not running the constructor (it doesn't actually do anything...).

In the future, I expect we'll want to move all of the logic (inside this `with_transaction` into some `send_external` function on the `CallManager`.

But up to you.

Reply:
(Really, I think we're going to need a bigger refactor in the future.